core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka
+
+import java.util
+import java.util.regex.Pattern
+
+import akka.Done
+import akka.actor.{Actor, ActorRef, Cancellable, Props, Status}
+import akka.event.LoggingReceive
+import org.apache.kafka.clients.consumer._
+import org.apache.kafka.common.TopicPartition
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Promise
+
+object KafkaConsumerActor {
+  case class StoppingException() extends Exception("Kafka consumer is stopping")
+  def props[K, V](settings: ConsumerSettings[K, V]) = Props(new KafkaConsumerActor(settings))
+
+  //requests
+  private[kafka] case class Assign(tps: Set[TopicPartition])
+  private[kafka] case class AssignWithOffset(tps: Map[TopicPartition, Long])
+  private[kafka] case class Subscribe(topics: Set[String], listener: ConsumerRebalanceListener)
+  private[kafka] case class SubscribePattern(pattern: String, listener: ConsumerRebalanceListener)
+  private[kafka] case class RequestMessages(topics: Set[TopicPartition])
+  private[kafka] case object Watch
+  private[kafka] case object Stop
+  private[kafka] case class Commit(offsets: Map[TopicPartition, Long])
+  //responses
+  private[kafka] case class Assigned(partition: List[TopicPartition])
+  private[kafka] case class Revoked(partition: List[TopicPartition])
+  private[kafka] case class Messages[K, V](messages: Iterator[ConsumerRecord[K, V]])
+  private[kafka] case class Committed(offsets: Map[TopicPartition, OffsetAndMetadata])
```

Comment:
Scala`s `private[x]` will be public in Java, by using a nested object you reduce the visible surface somewhat.